alphastats/gui/utils/ollama_utils.py

```diff
@@ -179,24 +181,39 @@ def execute_function(
             If the function is not implemented or the dataset is not available
         """
         try:
-            if function_name == "get_gene_function":
-                # TODO log whats going on
-                return get_gene_function(**function_args)
-            elif function_name == "get_enrichment_data":
-                return get_enrichment_data(**function_args)
-            elif function_name == "perform_dimensionality_reduction":
-                return perform_dimensionality_reduction(**function_args)
-            elif function_name.startswith("plot_") or function_name.startswith(
-                "perform_"
-            ):
+            # first try to find the function in the non-Dataset functions
+            if (
+                function := {
+                    "get_gene_function": get_gene_function,
+                    "get_enrichment_data": get_enrichment_data,
+                    "perform_dimensionality_reduction": perform_dimensionality_reduction,
+                }.get(function_name)
+            ) is not None:
+                return function(**function_args)
+
+            # special treatment for this one
+            elif function_name == "plot_intensity":
+                gene_name = function_args.pop("gene_name")
+                protein_id = get_gene_to_prot_id_mapping(
+                    gene_name, self._gene_to_prot_id_map
+                )
+                function_args["protein_id"] = protein_id
+
+                return self.dataset.plot_intensity(**function_args)
+
+            # fallback: try to find the function in the Dataset functions
```

Comment:
This whole logic is does not work as well anymore, now that we refactored most methods to no longer be directly available from dataset.

Reply:
note sure what you mean? all method calls should still work